### ISSUE TITLE
Re-visit adding a no-op `enable_fields_emulation` flag to 8.0

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#82485]
In #77749 (earlier PR  #75745) we introduced a flag that allows to emulates the "fields" option on CCS search requests going to pre-7.10 nodes where this option doesn't exist. With this flag set, we are trying to do a best-effort attempt at emulating this option
in mixed cluster or CCS scenarios. At the time, the flag was added only in version 7.16 and 7.17. The reasoning behind this was that since CCS from 8.0 is only supported to 7.17 (N-1) and that version fully supports the “fields” option, we would not need that flag on 8.0. 
I'm opening this issue since we might want to consider clients (e.g. Kibana) on 7.17 using this flag to be able to continue doing so in a rolling upgrade scenario, e.g. in a mixed ES 7.17/8.0 cluster. Currently, since we don't have the `enable_fields_emulation` flag on verison 8+, requests from a 7.17 client using it would fail because of an unknown rest parameter. 
We would need to at least allow the flag to be set (a no-op behaviour would be sufficient I think because 7.17 doesn't need "fields" emulation) and can later remove it.

**Comment by cbuescher** [comment]
I briefly talked to @stacey-gammon how this flag is e.g. currently used on the Kibana side. There might be no usage so far at all, in which case I don't think we should add an empty no-op flag to 8.0. However, if there should be some usage on their side at any point in the future we might need to add the flag on the 8+ releases depending on how expectations are what happens when a 7.last client using this flag talks to a 8+ node in a rolling upgrade.

**Comment by stacey-gammon** [comment]
Thanks @cbuescher for raising this. There may have been a miscommunication. It was my impression that the work being done on the Elasticsearch side, to implement the fields option compatibility layer, would Just Work in Kibana. I didn't realize we needed to set a flag to get it to work.

Because Kibana isn't using this flag, we might still have silent data loss in Discover under these circumstances. The good news is that I don't think we ever told customers it was fixed. Starting in 7.13 we documented it as a breaking change [here](https://www.elastic.co/guide/en/kibana/7.12/release-notes-7.12.0.html#breaking-changes-7.12.0) (granted, it's unlikely users upgrading, especially to a version > 7.13, will read that note) and we have this [open issue](https://github.com/elastic/kibana/issues/94965) about it.

I guess the question now is, should Kibana set this flag in 7.17?

@kertal or @LeeDr - can one of you verify whether silent data loss is occurring in Discover under those circumstances? (can use a Kibana 7.17 communicating with ES 7.9 via CCS).

cc @giladgal @thomasneirynck

**Comment by LeeDr** [comment]
I can work on a test.

**Comment by LeeDr** [comment]
I currently have Kibana/Elasticsearch 7.17.0 with CCS to Elasticsearch 7.9.3.
What I'm seeing isn't ideal, but I don't think I'd call it "data loss".  Discover does show all the docs.  It just doesn't show all the fields from the 7.9.3 docs.  But if I query on a field which isn't shown in the 7.9.3 docs I still get all the hits.

In my case I wrote docs to the remove 7.9.3 cluster with makelogs utility and let CCR replicate them to the 7.17.0 cluster.  I have index patterns for each cluster individually and combined so I can verify a query `response: 200` returns the same count of docs for each and 2x for the combined index pattern.

![image](https://user-images.githubusercontent.com/13542669/149211766-8af9670c-7371-461c-a599-80497bca09a5.png)

And if I change this advanced setting;
```
Read fields from _source
When enabled will load documents directly from `_source`. This is soon going to be deprecated. When disabled, will retrieve fields via the new Fields API in the high-level search service.
Default: false
discover:searchFieldsFromSource
```

![image](https://user-images.githubusercontent.com/13542669/149212111-e47dab1f-8dd9-473a-9333-a6f0a4ed5d46.png)

With that change Discover looks completely normal to me;
![image](https://user-images.githubusercontent.com/13542669/149212613-414485a8-d7b7-46c8-837c-fec083123ce3.png)

**Comment by stacey-gammon** [comment]
> It just doesn't show all the fields from the 7.9.3 docs. 

Yea, this is the data loss problem. Apologies, I first thought it was missing documents, but looking at past issues, it shows up as missing field data. The following two SDHs show this problem being reported in the field, and your test results indicate this is still a problem in 7.17.
 - https://github.com/elastic/sdh-kibana/issues/2206
 - https://github.com/elastic/sdh-kibana/issues/1406

**Comment by cbuescher** [comment]
> Yea, this is the data loss problem.

I would really prefer if we wouldn't call this data loss, from an ES point of view this usually means we lost some documents on ingestion and this is clearly not the case here. What's happening is that the "fields" options isn't available on pre-7.10 nodes and instead of throwing errors in these cases we made the decision to silently ignore the option in those cases back when we added it. So its "missing some fields" in responses from pre-7.10 nodes via CCS really.
